Code/GraphMol/MarvinParse/MarvinDefs.cpp

```diff
@@ -63,11 +53,570 @@ ptree MarvinMolBase::toPtree() const {
+    //       x2="-10.7921" y2="1.1658"/> <atom id="a3" elementType="C"
+    //       x2="-10.7921" y2="-0.3744"/> <atom id="a8" elementType="O"
+    //       x2="-12.1257" y2="-1.1444" lonePair="2"/>
+    //   </atomArray>
+
+    // See which one we have
+
+    std::string atomID =
+        atomArray.get<std::string>("<xmlattr>.atomID", "UseLongForm");
+    if (atomID == "UseLongForm") {
+      // long form - each atom on a line
+
+      BOOST_FOREACH (boost::property_tree::ptree::value_type &v,
+                     molTree.get_child("atomArray")) {
+        if (v.first != "atom") {
+          continue;
+        }
+
+        auto *mrvAtom = new MarvinAtom();
+        this->pushOwnedAtom(mrvAtom);
+        this->atoms.push_back(mrvAtom);
+
+        mrvAtom->id = v.second.get<std::string>("<xmlattr>.id", "");
+        mrvAtom->elementType =
+            v.second.get<std::string>("<xmlattr>.elementType", "");
+
+        if (mrvAtom->id == "" || mrvAtom->elementType == "") {
+          throw FileParseException(
+              "Expected id, elementType for an atom definition in MRV file");
+        }
+
+        std::string x2 = v.second.get<std::string>("<xmlattr>.x2", "");
+        std::string y2 = v.second.get<std::string>("<xmlattr>.y2", "");
+
+        // x2 and y2 are doubles
+
+        if (x2 != "" && y2 != "" &&
+            (!getCleanDouble(x2, mrvAtom->x2) ||
+             !getCleanDouble(y2, mrvAtom->y2))) {
+          throw FileParseException(
+              "The values x2 and y2 must be large floats in MRV file");
+        }
+
+        std::string formalCharge =
+            v.second.get<std::string>("<xmlattr>.formalCharge", "");
+        if (formalCharge != "") {
+          if (!getCleanInt(formalCharge, mrvAtom->formalCharge)) {
+            throw FileParseException(
+                "The value for formalCharge must be an integer in MRV file");
+          }
+        } else {
+          mrvAtom->formalCharge = 0;
+        }
+
+        mrvAtom->radical = v.second.get<std::string>("<xmlattr>.radical", "");
+        if (mrvAtom->radical != "") {
+          if (!boost::algorithm::contains(
+                  marvinRadicalVals,
+                  std::vector<std::string>{mrvAtom->radical})) {
+            std::ostringstream err;
+            err << "The value for radical must be one of "
+                << boost::algorithm::join(marvinRadicalVals, ", ")
+                << " in MRV file";
+            throw FileParseException(err.str());
+          }
+        } else {
+          mrvAtom->radical = "";
+        }
+
+        std::string isotopeStr =
+            v.second.get<std::string>("<xmlattr>.isotope", "");
+        if (isotopeStr != "") {
+          if (!getCleanInt(isotopeStr, mrvAtom->isotope) ||
+              mrvAtom->isotope <= 0) {
+            std::ostringstream err;
+            err << "The value for isotope must be a positive number in MRV file";
+            throw FileParseException(err.str());
```

Comment:
You can probably just do :
```
throw FileParseException("The value for isotope must be a positive number in MRV file");
```

Reply:
These have all been changed in the cases where the error string is a static cstr